### ISSUE TITLE
Building filters are inconsistent and should consider building=no

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#3779]
### Expected behavior

1) Polygon layers that do not include buildings should be filtered for:

`building IS NULL OR building = 'no'`

2) This should be the same for layers on different zoom levels

3) The indexes should match

### Actual behavior

1) The filters do not consider `building = 'no'` and only check for `building IS NULL` when selecting features, however the "is_building" columns check for `building = 'no'`, eg:

`CASE WHEN building = 'no' OR building IS NULL THEN 'no' ELSE 'yes' END AS is_building`

Specifically, these layers are filtered for `building IS NULL` only
`landcover-low-zoom`
`water-areas`
`landcover-area-symbols`
`landuse-overlay`
`text-poly-low-zoom`

2) `landcover`  is not filtered for buildings

3) The indexes are set to check for `building IS NULL` only, and should be changed

indexes.sql:
```
CREATE INDEX planet_osm_polygon_nobuilding
  ON planet_osm_polygon USING GIST (way)
  WHERE building IS NULL;
```

indexes.yml:
```
  nobuilding:
    where: building IS NULL
```

**Comment by imagico** [comment]
The questions i see are:

* is if there is anything speaking against changing this universally to `building = 'no' OR building IS NULL`?
* is there any substantial reason why the landcover layer does not currently use a `building IS NULL` condition?

**Comment by kocio-pl** [comment]
As for `landcover-low-zoom` - see #3622 (this is only for performance reasons as I did understand it).

**Comment by jeisenbe** [comment]
Any comments on this one, perhaps from @pnorman? 

I would like to change all of these layers to use `building = 'no' OR building IS NULL` including `landcover`, and adjust the indexes to match.

**Comment by pnorman** [comment]
I'd be okay with it, but to see if it's worth it, how common are building=no on something we'd otherwise pick up? Should we transform building=no into nulls in the transforms?

**Comment by jeisenbe** [comment]
Per taginfo there are 9 259 ways and 97 relations with 'building=no' https://taginfo.openstreetmap.org/tags/building=no#combinations

Here are checks from some keys and tags that we render:

1) 3080 ways with `amenity` + `building=no` - https://overpass-turbo.eu/s/JEu 

- example: 195 ways with amenity=marketplace + building=no - https://overpass-turbo.eu/s/JEw

2) 323 ways with `landuse` + `building=no` - https://overpass-turbo.eu/s/JEx

-  example: 55 ways with landuse=retail + building=no - https://overpass-turbo.eu/s/JEr

3) 532 ways with `shop` + `building=no` - https://overpass-turbo.eu/s/JEy

- example: 57 ways with `shop=mall` + `building=no` - https://overpass-turbo.eu/s/JEs

4) 473 ways with `leisure` + `building=no` - https://overpass-turbo.eu/s/JEC

- example: 276 ways with `leisure=sports_centre` + `building=no` - https://overpass-turbo.eu/s/JED

**Comment by jeisenbe** [comment]
There are also a number of features that could be imported as polygons if mapped as a closed way, even though they lack a tag that should be considered an area feature.

If I made this overpass-turbo query correctly, there are up to 4217 ways with `building=no` which lack a polygon key. 

I didn't check how many of these are closed ways, but many likely are (otherwise why use building=no?) -  https://overpass-turbo.eu/s/JEG

I'm not sure how this should be handled. Can we add `building=no` to `local linestring_values` in `openstreetmap-carto.lua` or would this improperly import some features as linestrings?

**Comment by jeisenbe** [comment]
> Should we transform building=no into nulls in the transforms?

I forgot about this idea. Should we implement this for v5.0.0? The alternative is to just use `building = 'no' OR building IS NULL` in the SQL queries. 

I suppose doing it in lua will make the import a little slower, but the rendering a little faster. Then again, it will make more sense if done in the transforms.

**Comment by jeisenbe** [comment]
It looks like this was already fixed by #4025
